src/algebra/group/units.lean

```diff
@@ -118,15 +130,13 @@ ext hv
 
 /-- Units of a monoid form a group. -/
 @[to_additive "Additive units of an additive monoid form an additive group."] instance : group αˣ :=
-{ mul := λ u₁ u₂, ⟨u₁.val * u₂.val, u₂.inv * u₁.inv,
-    by rw [mul_assoc, ← mul_assoc u₂.val, val_inv, one_mul, val_inv],
-    by rw [mul_assoc, ← mul_assoc u₁.inv, inv_val, one_mul, inv_val]⟩,
-  one := ⟨1, 1, one_mul 1, one_mul 1⟩,
-  mul_one := λ u, ext $ mul_one u,
-  one_mul := λ u, ext $ one_mul u,
+{ inv := has_inv.inv,
+  mul_left_inv := λ u, ext u.inv_val,
   mul_assoc := λ u₁ u₂ u₃, ext $ mul_assoc u₁ u₂ u₃,
-  inv := has_inv.inv,
-  mul_left_inv := λ u, ext u.inv_val }
+  mul := mul_one_class.mul,
+  one := mul_one_class.one,
+  one_mul := mul_one_class.one_mul,
+  mul_one := mul_one_class.mul_one}
```

Comment:
(side note: Isn't there be a shorter notation for these 4 lines, like `..mul_one_class`?)